tools/c7n_gcp/c7n_gcp/resources/compute.py

```diff
@@ -184,3 +184,122 @@ def get_resource_params(self, m, r):
         # Docs are wrong :-(
         # https://cloud.google.com/compute/docs/reference/rest/v1/snapshots/delete
         return {'project': project, 'snapshot': snapshot_id}
+
+
+@resources.register('security-policy')
+class SecurityPolicy(QueryResourceManager):
+    """GCP resource: https://cloud.google.com/compute/docs/reference/rest/v1/securityPolicies"""
+    class resource_type(TypeInfo):
+        service = 'compute'
+        version = 'v1'
+        component = 'securityPolicies'
+        id = 'name'
+
+        @staticmethod
+        def get(client, resource_info):
+            return client.execute_command(
+                'get', {'project': resource_info['project_id'],
+                        'securityPolicy': resource_info['policy_name']})
+
+
+@SecurityPolicy.action_registry.register('delete')
+class SecurityPolicyDelete(MethodAction):
+    """
+    `Deletes <https://cloud.google.com/compute/docs/reference/rest/v1/securityPolicies/delete>`_
+    a security policy
+
+    :Example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: security-policy-delete
+            description: Deletes a security policy
+            resource: gcp.security-policy
+            filters:
+              - type: value
+                key: name
+                value: test-policy
+            actions:
+              - delete
+    """
+
+    schema = type_schema('delete')
+    method_spec = {'op': 'delete'}
+    path_param_re = re.compile('.*?/projects/(.*?)/global/securityPolicies/(.*)')
+
+    def get_resource_params(self, m, r):
+        project, policy = self.path_param_re.match(r['selfLink']).groups()
+        return {'project': project, 'securityPolicy': policy}
+
+
+@SecurityPolicy.action_registry.register('add-rule')
```

Comment:
modify over add/remove naming so it can be extended in the future.

Reply:
done